pkg/endpoint/regenerationcontext.go

```diff
@@ -21,12 +21,43 @@ import (
 	"github.com/cilium/cilium/pkg/revert"
 )
 
+// DatapathRegenerationLevel determines what is expected of the datapath when
+// a regeneration event is processed.
+type DatapathRegenerationLevel int
+
+const (
+	// RegenerateWithoutDatapath indicates that datapath rebuild or reload
+	// is not required to implement this regeneration.
+	RegenerateWithoutDatapath DatapathRegenerationLevel = 0
```

Comment:
```
 const (
        // RegenerateWithoutDatapath indicates that datapath rebuild or reload
        // is not required to implement this regeneration.
-       RegenerateWithoutDatapath DatapathRegenerationLevel = 0
+       RegenerateWithoutDatapath DatapathRegenerationLevel = iota
        // RegenerateWithDatapathLoad indicates that the datapath must be
        // reloaded but not recompiled to implement this regeneration.
-       RegenerateWithDatapathLoad DatapathRegenerationLevel = 1
+       RegenerateWithDatapathLoad
        // RegenerateWithDatapathRebuild indicates the the datapath must be
        // recompiled and reloaded to implement this regeneration.
-       RegenerateWithDatapathRebuild DatapathRegenerationLevel = 2
+       RegenerateWithDatapathRebuild
 )
```

Reply:
```
$ git diff
diff --git a/pkg/endpoint/regenerationcontext.go b/pkg/endpoint/regenerationcontext.go
index d093dd9af0c3..b339d8c5aee2 100644
--- a/pkg/endpoint/regenerationcontext.go
+++ b/pkg/endpoint/regenerationcontext.go
@@ -28,13 +28,13 @@ type DatapathRegenerationLevel int
 const (
        // RegenerateWithoutDatapath indicates that datapath rebuild or reload
        // is not required to implement this regeneration.
-       RegenerateWithoutDatapath DatapathRegenerationLevel = 0
+       RegenerateWithoutDatapath DatapathRegenerationLevel = iota
        // RegenerateWithDatapathLoad indicates that the datapath must be
        // reloaded but not recompiled to implement this regeneration.
-       RegenerateWithDatapathLoad DatapathRegenerationLevel = 1
+       RegenerateWithDatapathLoad DatapathRegenerationLevel
        // RegenerateWithDatapathRebuild indicates the the datapath must be
        // recompiled and reloaded to implement this regeneration.
-       RegenerateWithDatapathRebuild DatapathRegenerationLevel = 2
+       RegenerateWithDatapathRebuild DatapathRegenerationLevel
 )
 
 // String converts a DatapathRegenerationLevel into a human-readable string.
$ make -C daemon
  CHECK contrib/scripts/bindata.sh
  GO    daemon/cilium-agent
# github.com/cilium/cilium/pkg/endpoint
../pkg/endpoint/regenerationcontext.go:34:29: const declaration cannot have type without expression
../pkg/endpoint/regenerationcontext.go:37:32: const declaration cannot have type without expression
make: *** [Makefile:15: cilium-agent] Error 2
```

(/cc @ianvernon who suggested the same thing)

Reply:
Oh, I see, the type is what screws it up. If I apply your diff it's fine. Thanks, I will roll this in and rebase.